src/modules/articles/components/ArticleBody.tsx

```diff
@@ -109,37 +108,71 @@ interface IProps {
+    super(props);
 
-  return (
-    <Row>
-      <Col xs={12} sm={12} md={8} lg={8} className={classes.ArticleBody}>
-        {SPEC_IMG_CAROUSEL_PATTERN.test(article.content) && (
-          <Lightbox title={article.title}>
-            <Gallery media={article.media} />
-          </Lightbox>
-        )}
-        {article.media && article.media.length > 0 && (
-          <ArticleFeaturedMedia
-            image={article.media[0]}
-            isCarouselButtonVisible={isCarouselButtonVisible}
-            carouselImageCount={article.media.length}
-          />
-        )}
-        <ArticleReference article={article} />
-        <div
-          className={classes.innerHTML}
-          dangerouslySetInnerHTML={{ __html: article.content }}
-        />
-      </Col>
-      <Col xsHidden smHidden mdOffset={1} md={3} lgOffset={1} lg={3}>
-        <RightRail />
-      </Col>
-    </Row>
-  );
-};
+    //creates div in which to store article contents
+    this.articleContentDiv = document.createElement("div");
+
+    this.createArticleContent = this.createArticleContent.bind(this);
+  }
+
+  private createArticleContent(element: HTMLDivElement | null): any {
+    //React gives us a DOM element from the render() method if it's been mounted.
+    //We fill this in with article content if it's not null.
+    if (element != null) {
+      element.appendChild(this.articleContentDiv)
+    }
+  }
+
+  private articleContentDiv: HTMLDivElement;
+
+  render() {
+    //Sets the div's contents to hold the HTML provided by the server.
+    this.articleContentDiv.innerHTML = this.props.article.content;
+
+    const { article, classes } = this.props;
+    const isFeaturedGalleryActive =
+      article.media && article.media.length > 0 && SPEC_IMG_CAROUSEL_PATTERN.test(article.content);
+    const featuredMedia = isFeaturedGalleryActive ? article.media : (article.media ? article.media.slice(0, 1) : undefined);
+
+    //Here we loop through the contents of the article div to find every element of tag type article-extension.
+    //We then create an Extension for each instance to render extension content inside the element.
+    const extensions = [];
+    const extensionElements = this.articleContentDiv.getElementsByTagName("article-extension");
+    if (extensionElements) {
+      for (let i = 0; i < extensionElements.length; i++) {
+        const type = extensionElements[i].getAttribute("type");
+        const props = extensionElements[i].getAttribute("props");
+        if (type && props) {
+          extensions.push(<Extension type={type} props={props} article={article} root={extensionElements[i]} />)
+        }
+      }
+    }
+
+    return (
+      <Row>
+        <Col xs={12} sm={12} md={8} lg={8} className={classes.ArticleBody}>
+          {featuredMedia && (
+            <ArticleMedia article={article} media={featuredMedia} />
+          )}
+          <ArticleReference article={article} />
+          <div
+            className={classes.innerHTML}
```

Comment:
`innerHTML` isn't an amazing name. How about `articleContent`?

Reply:
That's a good idea, all the jss stuff isn't mine but I'm glad to change it.